lib/iris/common/resolve.py

```diff
@@ -0,0 +1,1541 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+
+from collections import namedtuple
+from collections.abc import Iterable
+import logging
+
+from dask.array.core import broadcast_shapes
+
+from iris.common import LENIENT
+
+
+__all__ = ["Resolve"]
```

Comment:
Why is this part of the public API? Why not just make it private for now?

Reply:
@lbdreyer Simply because `Resolve` deprecates other functionality within `iris`, hence it requires to be part of the public API.

Also, it's functionality is quite generic and not necessarily ties to cube arithmetic. It provides a powerful way to combine two cubes into one resultant cube.